Makefile

```diff
@@ -34,7 +34,15 @@ all: test
 
 publish: clean ci compile build push
 
-ci: check-headers gofmt govet golint test
+ci: check-headers check-unused gofmt govet golint test
+
+.PHONY: check-unused
+check-unused:
+	@echo "==> Running check for unused/missing packages in go.mod"
+	@GO111MODULE=on go mod tidy
+	@echo "==> Running check for unused packages in vendor/"
+	@GO111MODULE=on go mod vendor
+	@git diff --exit-code -- go.sum go.mod vendor/
```

Comment:
I think it'd helpful to output a message to the user indicating how a failure of this check can be addressed. That is, I'd suggest to expand the line like (warning, untested)

```
@git diff --exit-code -- go.sum go.mod vendor/ || { echo "there are uncommitted changes to the Go modules and/or vendor files -- please run 'make vendor' and commit the changes first"; exit 1 }
```